master/internal/api_config_policies_intg_test.go

```diff
@@ -414,3 +544,1893 @@ func setupConfigPolicyAuthZ() *mocks.ConfigPolicyAuthZ {
+			configPolicies: map[string]any{
+				"invariant_config": map[string]interface {
+				}{
+					"description": "test\nspecial\tchar",
+					"environment": map[string]interface{}{
+						"force_pull_image": false,
+						"add_capabilities": []interface{}{"cap1", "cap2"},
+					},
+					"resources": map[string]interface{}{
+						"slots": float64(1),
+					},
+					"name": "my_experiment_config",
+				},
+				"constraints": map[string]interface{}{
+					"resources": map[string]interface{}{
+						"max_slots": float64(4),
+					},
+					"priority_limit": float64(10),
+				},
+			},
+			updatedPoliciesReq: &apiv1.PutGlobalConfigPoliciesRequest{
+				WorkloadType: model.ExperimentType,
+				ConfigPolicies: `{
+					"invariant_config": {
+						"description": "test\nspecial\tchar",
+						"environment": {
+							"add_capabilities": ["cap1"]
+						},
+						"resources": {
+							"slots": 5
+						},
+						"name": "my_experiment_config",
+						"entrypoint": "start from here",
+					},
+					"constraints": {
+						"resources": {
+							"max_slots": 8
+						},
+						"priority_limit": 10
+					}
+				}`,
+			},
+			updatedConfigPolicies: map[string]any{
+				"invariant_config": map[string]interface {
+				}{
+					"description": "test\nspecial\tchar",
+					"environment": map[string]interface{}{
+						"add_capabilities": []interface{}{"cap1"},
+					},
+					"resources": map[string]interface{}{
+						"slots": float64(5),
+					},
+					"name":       "my_experiment_config",
+					"entrypoint": "start from here",
+				},
+				"constraints": map[string]interface{}{
+					"resources": map[string]interface{}{
+						"max_slots": float64(8),
+					},
+					"priority_limit": float64(10),
+				},
+			},
+			err: nil,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			_, err := api.DeleteGlobalConfigPolicies(ctx,
+				&apiv1.DeleteGlobalConfigPoliciesRequest{WorkloadType: test.req.WorkloadType})
+
+			resp, err := api.PutGlobalConfigPolicies(ctx, test.req)
+			if test.err != nil {
+				require.ErrorContains(t, err, test.err.Error())
+				require.Nil(t, resp)
+				return
+			}
+			require.Nil(t, err)
+
+			// Verify that we can retrieve the input config policies.
+			getResp, err := api.GetGlobalConfigPolicies(ctx,
+				&apiv1.GetGlobalConfigPoliciesRequest{
+					WorkloadType: test.req.WorkloadType,
+				})
+			require.NoError(t, err)
+
+			configPolicies := getResp.ConfigPolicies.AsMap()
+			require.Equal(t, test.configPolicies, configPolicies)
+
+			resp, err = api.PutGlobalConfigPolicies(ctx, test.updatedPoliciesReq)
+			require.Nil(t, err)
+
+			// Verify that config policies were updated correctly.
+			getResp, err = api.GetGlobalConfigPolicies(ctx,
+				&apiv1.GetGlobalConfigPoliciesRequest{
+					WorkloadType: test.req.WorkloadType,
+				})
+			require.NoError(t, err)
+
+			updatedConfigPolicies := getResp.ConfigPolicies.AsMap()
+			require.Equal(t, test.updatedConfigPolicies, updatedConfigPolicies)
+		})
```

Comment:
That's a good point! I think in certain cases, longer test bodies could indicate the need for a helper function in the code, however I don't think this is one of those cases! The difference between an added config policy and an updated config policy is the matter of an `ON CONFLICT DO UPDATE` in the postgres code at the data layer, so this logic isn't handled in the API layer.
I prefer to leave this as is, since like you said, the table-driven approach is commonly used throughout our code base!